### PR TITLE
Update create space api

### PR DESCRIPTION
**Pull request by Dhwaniartefact**
Connected to https://github.com/archivematica/Issues/issues/1485

### CODE REVIEW

File: dev-manual/api/api-reference-storage-service.rst [hunk 1]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+
+``user``                A username for the DSpace instance with sufficient permissions to permit
+                        authentication.
+
+``password``            The password for the Dspace user that you created.
+
+``metadata_policy``     Use to restrict access to the metadata bitstream. Must be specified as a list of
+                        objects in JSON, e.g. ``[{"action":"READ","groupId":"5","rpType":"TYPE_CUSTOM"}]``.
+
+``archive_format``      Select Archive format from dropdown. e.g. ``ZIP or 7z``
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]: What dropdown does this refer to? Would it be enough to just list the values of the field choices?

```suggestion
``archive_format``      Archive format, e.g. ``ZIP``, ``7z``.
```

File: dev-manual/api/api-reference-storage-service.rst [hunk 2]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+
+``as_user``               ArchivesSpace username to authenticate as.
+
+``as_password``           ArchivesSpace password to authenticate with.
+
+``as_repository``         Identifier of the default ArchivesSpace repository.
+
+``as_archival_object``    Identifier of the default ArchivesSpace archival object.
+
+``upload_to_tsm``         Boolean True or False. Default is False. will attempt to send the AIP to
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]: If the content type of this endpoint is `json`, these boolean values are incorrect.

```suggestion
``upload_to_tsm``         Boolean true or false. Default is false. will attempt to send the AIP to
```

File: dev-manual/api/api-reference-storage-service.rst [hunk 3]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+                          the Tivoli Storage Manager using command-line client dsmc
+
+``verify_ssl``            Boolean True or False. Default is True.
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]:
```suggestion
``verify_ssl``            Boolean true or false. Default is true.
```

File: dev-manual/api/api-reference-storage-service.rst [hunk 4]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+========================    ===============================================================
+
+``sd_iri``                  The URL of the LOCKSS-o-matic service document IRI,
+                            e.g. ``http://lockssomatic.example.org/api/sword/2.0/sd-iri``.
+
+``content_provider_id``     On-Behalf-Of value when communicating with LOCKSS-o-matic.
+
+``external_domain``         Base URL for this server that LOCKSS will be able to access.
+
+``keep_local``              Boolean True or False. Default is True.
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]: Is the endpoint content type `json`?

File: dev-manual/api/api-reference-storage-service.rst [hunk 5]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+
+====================    ================================================
+
+``remote_name``         Name of the NFS server.
+
+``remote_path``         Path on the NFS server to the export.
+
+``version``             Type of the filesystem, e.g. ``nfs or nfs4``
+
+``manually_mounted``    Boolean True or False. Default is False.
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]: Is the endpoint content type `json`?

File: dev-manual/api/api-reference-storage-service.rst [hunk 6]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+
+Request body parameters:
+
+========================   ================================================
+
+``remote_user``            The username on the remote host.
+
+``remote_name``            Name or IP of the remote machine.
+
+``assume_rsync_daemon``    Boolean True or False. Default is False.
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]: Is the endpoint content type `json`?

File: dev-manual/api/api-reference-storage-service.rst [hunk 7]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+^^^^^^^^^^^^^^^^
+
+Archipelago
+^^^^^^^^^^^
+
+Archipelago is currently supported for AIP Storage locations.
+Archipelago is built on the content management system Drupal,
+which accepts JSON API calls to create new entities.
+Basic fields for a local file system space are listed `above \
+<https://www.archivematica.org/en/docs/archivematica-1.15/dev-manual/api/api-reference-storage-service/#create-space>`_
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]: Should we make the links in the descriptions of each access protocol [internal](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#internal-links)? That way we don't need to update them on each release (notice how some of the existing links in this page still point to the Storage Service 0.18 docs).

If for some reason you still need to use full links to the docs, using `latest` in the version (e.g `archivematica-1.15`) could be more maintainable.

File: dev-manual/api/api-reference-storage-service.rst [hunk 8]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+``agent_type``          A string that will be used in Archivematica METS file to
+                        identify the type of PREMIS agent named above,
+                        e.g. organization.Name or IP of the remote machine.
+
+``agent_identifier``    A string that will be used as an identifier in the Archivematica
+                        METS file to uniquely identify the PREMIS agent named above.
+
+=====================   ===============================================================
+
+Duracloud
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst]:
```suggestion
DuraCloud
```

File: dev-manual/api/api-reference-storage-service.rst [hunk 9]
```diff
@@ -540,13 +544,330 @@ Where payload.json contains::
...
+
+``fedora_user``          Fedora user name (for SWORD functionality).
+
+``fedora_password``      Fedora password (for SWORD functionality).
+
+``fedora_name``          Name or IP of the remote Fedora machine.
+
+====================     ==================================================
+
+LOCKSS
```
Comment (replaceafill) [inline dev-manual/api/api-reference-storage-service.rst:756]: Any reason you skipped the `GPG encryption` space?

**Review by replaceafill — APPROVED** [review]
This looks good @Dhwaniartefact. I think you only need to consider the use of `True` and `False` as boolean parameters. I also noticed the `GPG encryption` space is missing, not sure why.

File: dev-manual/api/api-reference-storage-service.rst (hunk 7 above)
Comment (Dhwaniartefact) [reply dev-manual/api/api-reference-storage-service.rst]: Thanks for suggestion. Let me make changes accordingly.

File: dev-manual/api/api-reference-storage-service.rst (hunk 9 above)
Comment (Dhwaniartefact) [reply dev-manual/api/api-reference-storage-service.rst:756]: Not really, let me add soon.

**Pull request merged**